### ISSUE TITLE
Support for in house oauth server

### ISSUE THREAD

**Issue by bizzerk212** [exceptionless/Exceptionless#493]
TLDR: If I have an in house oauth2 server, can I use that to login to a self hosted exceptionless server?

I am in kind of a interesting situation.  My office uses mostly linux server, except for a couple of windows servers for hosting the active directory domain and running applications that absolutely cannot run on a linux server.  Needless to say, the windows servers are at a premium and I don't want to use one if I can avoid it.  To allow for domain authentication of our internal web applications, we built an oauth2 server that runs on windows and authenticates against the domain.  This way we have control over which users have access to which applications and our applications running on linux servers (which is most of them) can authenticate with the domain (since, at this time, the domain services on dotnet core are only supported on windows servers:  https://github.com/dotnet/corefx/issues/24843).  
The question is, can I use the exceptionless config (EX_ConnectionStrings__OAuth) to connect to my in house oauth server?  If not, could that feature be added?

I know there was an issue submitted around connecting exceptionless to active directory on linux, but I am really just asking if I can connect it to my oauth server instead of having to login with google/facebook/github/microsoft.  Would it be possible to connect exceptionless to basically any oauth2 server I want, including one running in house?

**Comment by niemyjski** [comment]
No, you'd need to add both client and server side support to talk to your oauth server. We'd be willing to accept a pr that reads existing oauth configuration on both the client and server side to talk to a predefined oauth2 server. I don't know how much work this would be, but I don't think it would be that bad considering all the infrastructure is in place, you'd need to look at the ui usage as well as the auth controller server side for adding your own configuration. We could reuse that connection string, but you'd need to build a custom oauth endpoint and then update the client to call it.

I'm going to close this for now, but please feel free to open a pr for this!

**Comment by niemyjski** [comment]
Also, if you need any help getting started with this, please let us know. I'd be more than happy to get you up and running.

**Comment by bizzerk212** [comment]
Ok, thank you.  I will read through the contributing documentation and pull down the source code.  What is the best why of getting in contact with you/the rest of the community if I have questions about the code?

**Comment by niemyjski** [comment]
Our slack channel which is in the readme.
